example/fly_mission/fly_mission_dronecore.cpp

```diff
@@ -0,0 +1,357 @@
+                                                 bool is_fly_through,
+                                                 float gimbal_pitch_deg,
+                                                 float gimbal_yaw_deg,
+                                                 MissionItem::CameraAction camera_action);
+
+
+int main(int /*argc*/, char ** /*argv*/)
+{
+    DroneCore dc;
+
+    bool discovered_device = false;
+    // Discover and connect to drone
+    DroneCore::ConnectionResult connection_result = dc.add_udp_connection();
+
+    if (connection_result != DroneCore::ConnectionResult::SUCCESS)
+    {
+        std::cout << DRONECORE << "Connection failed: " << DroneCore::connection_result_str(
+                                                               connection_result)
+                  << std::endl;
+        return 1;
+    }
+
+    std::cout << DRONECORE << "Waiting to discover device..." << std::endl;
+    dc.register_on_discover([&discovered_device](uint64_t uuid) {
+        std::cout << DRONECORE << "Discovered device with UUID: " << uuid << std::endl;
+        discovered_device = true;
+    });
+
+    // We usually receive heartbeats at 1Hz, therefore we should find a device after around 2 seconds.
+    std::this_thread::sleep_for(std::chrono::seconds(2));
+    if (!discovered_device)
+    {
+        std::cout << DRONECORE << "No device found, exiting." << std::endl;
+        return 1;
+    }
+
+    // We don't need to specifiy the UUID if it's only one device anyway.
+    // If there were multiple, we could specify it with:
+    // dc.device(uint64_t uuid);
+    Device &device = dc.device();
+	std::vector<std::shared_ptr<MissionItem>> mission_items;
+
+    mission_items.push_back(
+        add_waypoint(47.398170327054473,
+                     8.5456490218639658,
+                     10.0f, 5.0f, false,
+                     20.0f, 60.0f,
+                     MissionItem::CameraAction::NONE));
+
+    mission_items.push_back(
+        add_waypoint(47.398241338125118,
+                     8.5455360114574432,
+                     10.0f, 2.0f, true,
+                     0.0f, -60.0f,
+                     MissionItem::CameraAction::TAKE_PHOTO));
+
+    mission_items.push_back(
+        add_waypoint(47.398139363821485, 8.5453846156597137,
+                     10.0f, 5.0f, true,
+                     -46.0f, 0.0f,
+                     MissionItem::CameraAction::START_VIDEO));
+
+    mission_items.push_back(
+        add_waypoint(47.398058617228855,
+                     8.5454618036746979,
+                     10.0f, 2.0f, false,
+                     -90.0f, 30.0f,
+                     MissionItem::CameraAction::STOP_VIDEO));
+
+    mission_items.push_back(
+        add_waypoint(47.398100366082858,
+                     8.5456969141960144,
+                     10.0f, 5.0f, false,
+                     -45.0f, -30.0f,
+                     MissionItem::CameraAction::START_PHOTO_INTERVAL));
```

Comment:
1s according to:
https://github.com/dronecore/DroneCore/blob/1f0e92723e6c85de2d2a069ac3d4436dc9b8028e/plugins/mission/mission_impl.cpp#L353
https://github.com/dronecore/DroneCore/blob/5110ab992871e8f4b254295a70d955e9f178efe9/plugins/mission/mission_item_impl.h#L30
https://github.com/dronecore/DroneCore/blob/5110ab992871e8f4b254295a70d955e9f178efe9/plugins/mission/mission_item_impl.h#L53